symja_android_library/matheclipse-core/src/main/java/org/matheclipse/core/expression/NumberUtil.java

```diff
@@ -154,48 +154,23 @@ public static boolean isOdd(BigInteger a) {
     return a.testBit(0);
   }
 
-  public static long toLong(BigInteger a) throws ArithmeticException {
-    long val = a.longValue();
-    if (!a.equals(BigInteger.valueOf(val))) {
-      throw new ArithmeticException("BigInteger:toLong: number out of range");
-    }
-    return val;
+  public static boolean hasIntValue(BigInteger value) {
+    // This is simlar to what is checked in BigInteger.intValueExact()
+    return value.bitLength() <= 31;
   }
 
-  /**
-   * Converts this number to <code>int</code>. If this integer cannot be represented by an <code>int
-   * </code> type, return the <code>Integer.MIN_VALUE</code> .
-   *
-   * @return the numeric value represented by this integer after conversion to type <code>int</code>
-   *         or the <code>Integer.MIN_VALUE</code> if the cast is not possible
-   */
-  public static int toIntDefault(BigInteger a) {
-    return toIntDefault(a, Integer.MIN_VALUE);
+  public static boolean hasLongValue(BigInteger value) {
+    // This is simlar to what is checked in BigInteger.longValueExact()
+    return value.bitLength() <= 63;
   }
 
-  /**
-   * Converts this number to <code>int</code>. If this integer cannot be represented by an <code>int
-   * </code> type, return the <code>defaultValue</code> .
-   *
-   * @return the numeric value represented by this integer after conversion to type <code>int</code>
-   *         or the <code>defaultValue</code> if the cast is not possible
-   */
-  public static int toIntDefault(BigInteger a, final int defaultValue) {
-    int val = a.intValue();
-    if (!a.equals(BigInteger.valueOf(val))) {
-      return defaultValue;
-      // throw new ArithmeticException("BigInteger:toInt: number out of range");
+  public static long toLong(BigInteger a) throws ArithmeticException {
+    // TODO: use the following?
+    a.longValueExact();
```

Comment:
@axkr what do you think about just using BigInteger.longValueExact() here.
It throws the same type of exception, but with a slightly different message that carries IMHO the same information:
It throws: `new ArithmeticException("BigInteger out of long range")`

Reply:
yes use `BigInteger.longValueExact()`

Reply:
Done.